x-pack/plugins/uptime/server/lib/requests/get_journey_screenshot_blocks.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isRight } from 'fp-ts/lib/Either';
+import * as t from 'io-ts';
+import { ScreenshotBlockDoc } from '../../../common/runtime_types';
+import { UMElasticsearchQueryFn } from '../adapters/framework';
+
+const ScreenshotBlockResultType = t.array(
+  t.type({
+    _id: t.string,
+    _source: t.type({
+      synthetics: t.type({
+        blob: t.string,
+        blob_mime: t.string,
+      }),
+    }),
+  })
+);
+
+export const getJourneyScreenshotBlocks: UMElasticsearchQueryFn<
+  { blockIds: string[] },
+  ScreenshotBlockDoc[]
+> = async ({ blockIds, uptimeEsClient }) => {
+  const body = {
+    query: {
+      bool: {
+        filter: [
+          {
+            ids: {
+              values: blockIds,
+            },
+          },
+        ],
+      },
+    },
+    size: 10000,
```

Comment:
how many images are covered in approx 10000 blocks? 

i am concerned about the size of data here we are fetching and returning, it can become really huge for a site with heavy animation going on.

Reply:
++ - we can limit it to some smaller size. IIRC 10k is default max doc set, so it was just force of habit. It's not some ungodly amount of data, I think roughly 12 MB. When I ran it locally and fetched that many, it took about 3400ms and the response size was ~11mb. 

If we limit it to 1000 docs, it takes 1mb and the query runs around 270ms for me. I think 1k is a good limit. A typical screenshot is about 70kb in size, so I think most requests will fall far short of max size. I guess we will want some intelligence on the client to make sure it doesn't attempt to request too many in a single request.

Reply:
@shahzad31 I reduced to 1k cap https://github.com/elastic/kibana/pull/101678/commits/b1f7576362240f00ab57c222c19b1593a9c9d7fd.